### PR TITLE
MM-40127 : Fix team sidebar buttons screen reader hover

### PR DESCRIPTION
**Pull request by jbattistispiria**
#### Summary
Fix screen reader does not read out buttons in team sidebar on hover.
Screen reader now say:
1. Team image: "{team name} Team Image"
2. Team initials: "{team name} Team Initials"
3. Plus icon: "Plus icon"

Can't fix team sidebar button on click because the focus is given by default to the post textbox after the page load.
So, screen reader read the post textbox aria-label.

Update `team_icon.test.tsx.snap` snapshot.

#### Ticket Link
Fix https://github.com/mattermost/mattermost-server/issues/19161
Jira: https://mattermost.atlassian.net/browse/MM-40127
#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: components/team_sidebar/team_sidebar.tsx [hunk 1]
```diff
@@ -259,6 +259,14 @@ export default class TeamSidebar extends React.PureComponent<Props, State> {
 
         const joinableTeams = [];
 
+        const plusIcon = (
+            <i
+                className='icon icon-plus'
+                role={'img'}
+                aria-label={'Plus Icon'}
```
Comment (M-ZubairAhmed) [inline components/team_sidebar/team_sidebar.tsx:266]: we localize the aria labels too, can we do that here

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Review by devinbinnie — APPROVED** [review]
LGTM other than @M-ZubairAhmed's comment. Thanks @jbattistispiria!

File: components/team_sidebar/team_sidebar.tsx (hunk 1 above)
Comment (jbattistispiria) [reply components/team_sidebar/team_sidebar.tsx]: Done. I also added localization in `team_icon.tsx`.

**Review by M-ZubairAhmed — APPROVED** [review]

**Comment by jgilliam17** [comment]
Thanks @jbattistispiria 
Tested, looks good to merge.
- Verified screen reader is reading out team buttons.
- [E2E report](https://mattermost-cypress-report.s3.amazonaws.com/188257-5f90cc3-onprem-ent-webapp-pr-10227-mattermost/mm-ee-test5f90cc3/mochawesome.html) looks good, no PR related failures.

**Review by jgilliam17 — APPROVED** [review]

**Pull request merged**